microprofile/opentracing-smallrye/src/main/java/org/wildfly/extension/microprofile/opentracing/smallrye/TracingDeploymentProcessor.java

```diff
@@ -0,0 +1,124 @@
+
+  @Override
+  public void undeploy(DeploymentUnit deploymentUnit) {
+  }
+
+  private void addListeners(DeploymentUnit deploymentUnit) {
+    JBossWebMetaData jbossWebMetaData = getJBossWebMetaData(deploymentUnit);
+    if (null == jbossWebMetaData) {
+      // nothing to do here
+      return;
+    }
+
+    TracingLogger.ROOT_LOGGER.registeringTracerInitializer();
+
+    String serviceName = getServiceName(deploymentUnit);
+    ParamValueMetaData serviceNameContextParameter = new ParamValueMetaData();
+    serviceNameContextParameter.setParamName(TracerInitializer.SMALLRYE_OPENTRACING_SERVICE_NAME);
+    serviceNameContextParameter.setParamValue(serviceName);
+    addContextParameter(jbossWebMetaData, serviceNameContextParameter);
+
+    ListenerMetaData listenerMetaData = new ListenerMetaData();
+    listenerMetaData.setListenerClass(TracerInitializer.class.getName());
+
+    List<ListenerMetaData> listeners = jbossWebMetaData.getListeners();
+    if (null == listeners) {
+      listeners = new ArrayList<>();
+    }
+    listeners.add(listenerMetaData);
+    jbossWebMetaData.setListeners(listeners);
+  }
+
+  private void addJaxRsIntegration(DeploymentUnit deploymentUnit) {
+    JBossWebMetaData jbossWebMetaData = getJBossWebMetaData(deploymentUnit);
+    if (null == jbossWebMetaData) {
+      // nothing to do here
+      return;
+    }
+
+    TracingLogger.ROOT_LOGGER.registeringJaxRs();
+
+    ParamValueMetaData restEasyProvider = new ParamValueMetaData();
+    restEasyProvider.setParamName("resteasy.providers");
+    restEasyProvider.setParamValue(SmallRyeTracingDynamicFeature.class.getName());
+    addContextParameter(jbossWebMetaData, restEasyProvider);
+  }
+
+  private void addCDIExtension(DeploymentUnit deploymentUnit) {
+    TracingLogger.ROOT_LOGGER.registeringCDIExtension();
+
+    WeldPortableExtensions extensions = WeldPortableExtensions.getPortableExtensions(deploymentUnit);
+    extensions.registerExtensionInstance(new TracingCDIExtension(), deploymentUnit);
+  }
+
+  private JBossWebMetaData getJBossWebMetaData(DeploymentUnit deploymentUnit) {
+    WarMetaData warMetaData = deploymentUnit.getAttachment(WarMetaData.ATTACHMENT_KEY);
+    if (null == warMetaData) {
+      // not a web deployment, nothing to do here...
+      return null;
+    }
+
+    return warMetaData.getMergedJBossWebMetaData();
+  }
+
+  private void addContextParameter(JBossWebMetaData jbossWebMetaData, ParamValueMetaData restEasyProvider) {
+    List<ParamValueMetaData> contextParams = jbossWebMetaData.getContextParams();
+    if (null == contextParams) {
+      contextParams = new ArrayList<>();
+    }
+    contextParams.add(restEasyProvider);
+    jbossWebMetaData.setContextParams(contextParams);
+  }
+
+  private String getServiceName(DeploymentUnit deploymentUnit) {
+    String serviceName = System.getProperty("JAEGER_SERVICE_NAME");
+    if (null == serviceName || serviceName.isEmpty()) {
+      serviceName = System.getenv("JAEGER_SERVICE_NAME");
```

Comment:
This should use `WildFlySecurityManager.getEnvPropertyPrivileged()`. Same comment about the risk as well.